packages/next-auth/src/providers/linkedin.ts

```diff
@@ -30,7 +30,25 @@ export default function LinkedIn<P extends LinkedInProfile>(
       url: "https://www.linkedin.com/oauth/v2/authorization",
       params: { scope: "r_liteprofile r_emailaddress" },
     },
-    token: "https://www.linkedin.com/oauth/v2/accessToken",
+    token: {
+      url: "https://www.linkedin.com/oauth/v2/accessToken",
+      async request({
+                      client,
+                      params,
+                      checks,
+                      provider
+                    }) {
+        const response = await client.oauthCallback(provider.callbackUrl, params, checks, {
+          exchangeBody: {
+            client_id: options.clientId,
+            client_secret: options.clientSecret,
+          }
+        });
+        return {
+          tokens: response
+        };
+      }
+    },
```

Comment:
This is not necessary, you can just add the following config:

```js
client: {
      token_endpoint_auth_method: "client_secret_post",
},
```

Read more here: https://next-auth.js.org/configuration/providers/oauth#client-option

Reply:
the solution from @balazsorban44 works for me as well, thanks